### PR TITLE
Update with the latest changes from rpk for 24.2.5

### PR DESCRIPTION
**Pull request by Deflaimun**
## Should only be merged after 24.2.5
Check https://github.com/redpanda-data/redpanda/releases to make sure the version is out. 

## Description

Adds the new changes extracted from rpk.

Disconsider the branch name. Should be 24.2.5
Resolves https://github.com/redpanda-data/documentation-private/issues/<add-your-issue-number-here>
Review deadline:

## Page previews

https://deploy-preview-775--redpanda-docs-preview.netlify.app/current/reference/rpk/rpk-cluster/rpk-cluster-partitions-balance/
https://deploy-preview-775--redpanda-docs-preview.netlify.app/current/reference/rpk/rpk-cluster/rpk-cluster-health/
https://deploy-preview-775--redpanda-docs-preview.netlify.app/current/reference/rpk/rpk-redpanda/rpk-redpanda-start/
https://deploy-preview-775--redpanda-docs-preview.netlify.app/current/reference/rpk/rpk-topic/rpk-topic-consume/
https://deploy-preview-775--redpanda-docs-preview.netlify.app/current/reference/rpk/rpk-topic/rpk-topic-produce/
https://deploy-preview-775--redpanda-docs-preview.netlify.app/current/reference/rpk/rpk-transform/rpk-transform-logs/

## Checks

- [x] New feature
- [ ] Content gap
- [ ] Support Follow-up
- [x] Small fix (typos, links, copyedits, etc)

### CODE REVIEW

**Review by r-vasquez — APPROVED** [review]
Thanks, LGTM!

File: modules/reference/pages/rpk/rpk-topic/rpk-topic-consume.adoc [hunk 1]
```diff
@@ -94,6 +94,12 @@ Headers are formatted with percent encoding inside of the modifier:
 %h{%k=%v{hex}}
 ```
 
+=== Values
+
+Values for consumed records can be omitted by using the `--meta-only` flag.
+
+Tombstone records (records with a `null` value) have their value omitted from the JSON output by default. All other records, including those with an empty-string value (`""``), will have their values printed.
```
Comment (WillemKauf) [inline modules/reference/pages/rpk/rpk-topic/rpk-topic-consume.adoc]: (\`""``) should be (\`""`)

File: modules/reference/pages/rpk/rpk-topic/rpk-topic-produce.adoc [hunk 2]
```diff
@@ -39,6 +39,24 @@ Produce to `foo`, using schema ID 1, message FQN Person.Name:
...
+```
+    
+Tombstone records (records with a `null` value) can be produced by using the `-Z` flag and creating empty-string value records. Using the same example from above, but adding the `-Z` flag will produce a record with the key `tombstone_record` and the value `null`:
+
+```bash
+rpk topic produce foo -k tombstone_record -Z 
+[Enter]
+```
+
+It is important to note that records produced with values of string `null` are not considered tombstones by Redpanda.
```
Comment (WillemKauf) [inline modules/reference/pages/rpk/rpk-topic/rpk-topic-produce.adoc]: This should read,

>It is important to note that records produced with values of string \`"null"` are not considered tombstones by Redpanda.


Note the quotations around null- very important distinction.

**Review by WillemKauf — CHANGES_REQUESTED** [review]

File: modules/reference/pages/rpk/rpk-topic/rpk-topic-consume.adoc (hunk 1 above)
Comment (Deflaimun) [inline modules/reference/pages/rpk/rpk-topic/rpk-topic-consume.adoc]:
```suggestion
Tombstone records (records with a `null` value) have their value omitted from the JSON output by default. All other records, including those with an empty-string value (`""`), will have their values printed.
```

File: modules/reference/pages/rpk/rpk-topic/rpk-topic-produce.adoc (hunk 2 above)
Comment (Deflaimun) [inline modules/reference/pages/rpk/rpk-topic/rpk-topic-produce.adoc]:
```suggestion
It is important to note that records produced with values of string `"null"` are not considered tombstones by Redpanda.
```

File: modules/reference/pages/rpk/rpk-topic/rpk-topic-consume.adoc [hunk 3]
```diff
@@ -94,6 +94,12 @@ Headers are formatted with percent encoding inside of the modifier:
 %h{%k=%v{hex}}
 ```
 
+=== Values
```
Comment (WillemKauf) [inline modules/reference/pages/rpk/rpk-topic/rpk-topic-consume.adoc]: This sub-section seems to have been placed in between the `Headers` sub-section. It needs to be moved below the paragraph:

```
This prints all headers with a space before the key and after the value, an equals sign between the key and value, and with the value hex encoded. Header formatting actually just parses the internal format as a record format, so all of the above rules about %K, %V, text, and numbers apply.
```

Comment (Deflaimun) [reply modules/reference/pages/rpk/rpk-topic/rpk-topic-consume.adoc]: Done! I've reorganized the headings too for better visibility.

**Review by WillemKauf — APPROVED** [review]
Thanks!

**Pull request merged**